WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -186,7 +186,9 @@ extension PointOfSaleAggregateModel {
 @available(iOS 17.0, *)
 extension PointOfSaleAggregateModel {
     func enableCoupons() async {
-        await couponsController.enableCoupons()
+        if await couponsController.enableCoupons() {
```

Comment:
I was going back and forth on whether `enableCoupons` should return a result, or throw an error. In the end, opted for the most straightforward solution - return a boolean. Since the error is handled within the controller, there's no point in propagating the error for the controller's caller.